packages/grafana-ui/src/components/Segment/types.ts

```diff
@@ -0,0 +1,15 @@
+import { ReactElement } from 'react';
+import { SelectableValue } from '@grafana/data';
+
+type ObjectOption<T> = {
+  [key: string]: Array<SelectableValue<T>>;
+};
+
+export type OptionType<T> = Array<SelectableValue<T>> | ObjectOption<T>;
```

Comment:
Can you explain why this is needed, can't grouped options just be passed in like an array of groups?

Reply:
definitely!